azkaban-common/src/test/java/azkaban/executor/ExecutionFlowDaoTest.java

```diff
@@ -176,6 +182,47 @@ public void testFetchQueuedFlows() throws Exception {
     assertTwoFlowSame(flow2, fetchedFlow2.getSecond());
   }
 
+  @Test
+  public void testAssignAndUnassignExecutor() throws Exception {
+    final String host = "localhost";
+    final int port = 12345;
+    final Executor executor = this.executorDao.addExecutor(host, port);
+    final ExecutableFlow flow = TestUtils.createExecutableFlow("exectest1", "exec1");
+    this.executionFlowDao.uploadExecutableFlow(flow);
+    this.assignExecutor.assignExecutor(executor.getId(), flow.getExecutionId());
+
+    final Executor fetchExecutor = this.executorDao.fetchExecutorByExecutionId(flow.getExecutionId());
+    assertThat(fetchExecutor).isEqualTo(executor);
+
+    this.assignExecutor.unassignExecutor(flow.getExecutionId());
+    assertThat(this.executorDao.fetchExecutorByExecutionId(flow.getExecutionId())).isNull();
+  }
+
+  /* Test exception when assigning a non-existent executor to a flow */
+  @Test
+  public void testAssignExecutorInvalidExecutor() throws Exception {
+    final ExecutableFlow flow = TestUtils.createExecutableFlow("exectest1", "exec1");
+    this.executionFlowDao.uploadExecutableFlow(flow);
+
+    assertThatThrownBy(
+        () -> this.assignExecutor.assignExecutor(flow.getExecutionId(), 1))
```

Comment:
Should it be assignExecutor(1, flow.getExecutionId())? Also how do you know 1 is a non-existent executor?